search-app/src/cmr/search/services/query_execution/facets/temporal_facets.clj

```diff
@@ -23,10 +23,32 @@
   [datetime interval]
   (second (re-find #"^\d{4}-\d{2}-\d{2}T(\d{2}):\d{2}:\d{2}\+\d+$" datetime)))
 
+(defn- time-intervals->next-interval
```

Comment:
I really like how you're helping to drive the CMR to smaller functions with single responsibilities :-)